x/stake/keeper/validator.go

```diff
@@ -246,13 +257,70 @@ func (k Keeper) UpdateValidator(ctx sdk.Context, validator types.Validator) type
 			bz := k.cdc.MustMarshalBinary(validator.ABCIValidator())
 			store.Set(GetTendermintUpdatesKey(validator.Owner), bz)
 		}
-
 	}
 
 	k.SetValidator(ctx, validator)
 	return validator
 }
 
+// updateCliffValidator determines if the current cliff validator needs to be
+// updated or swapped. If the provided affected validator is the current cliff
+// validator before it's power was increased, either the cliff power key will
+// be updated or if it's power is greater than the next bonded validator by
+// power, it'll be swapped.
+func (k Keeper) updateCliffValidator(ctx sdk.Context, affectedVal types.Validator) {
+	store := ctx.KVStore(k.storeKey)
+	pool := k.GetPool(ctx)
+	cliffAddr := sdk.AccAddress(k.GetCliffValidator(ctx))
+
+	oldCliffVal, found := k.GetValidator(ctx, cliffAddr)
+	if !found {
+		panic(fmt.Sprintf("cliff validator record not found for address: %v\n", cliffAddr))
+	}
+
+	// NOTE: We get the power via affectedVal since the store (by power key)
+	// has yet to be updated.
+	affectedValPower := affectedVal.GetPower()
+
+	// Create a validator iterator ranging from smallest to largest by power
+	// starting the current cliff validator's power.
+	start := GetValidatorsByPowerIndexKey(oldCliffVal, pool)
+	end := sdk.PrefixEndBytes(ValidatorsByPowerIndexKey)
+	iterator := store.Iterator(start, end)
+
+	var newCliffVal types.Validator
+
+	for ; iterator.Valid(); iterator.Next() {
```

Comment:
I still don't understand the point of a for loop with only one iteration, why do we need a loop structure at all?

Reply:
Yeah ++ this comment we shouldn't have a loop it should just panic at the higher level if _it would have_ reached the end of the loop

Reply:
Guys my brain was fried yesterday...sorry about this.

Reply:
haha no worries (me too!) thanks for the updated